module/zfs/abd.c

```diff
@@ -216,11 +223,52 @@ struct page;
 #define	kpm_enable			1
 #define	abd_alloc_chunk() \
 	((struct page *) umem_alloc_aligned(PAGESIZE, 64, KM_SLEEP))
-#define	abd_free_chunk(chunk) 		umem_free(chunk, PAGESIZE)
+#define	abd_free_chunk(chunk)		umem_free(chunk, PAGESIZE)
 #define	zfs_kmap_atomic(chunk, km)	((void *)chunk)
 #define	zfs_kunmap_atomic(addr, km)	do { (void)(addr); } while (0)
 #define	local_irq_save(flags)		do { (void)(flags); } while (0)
 #define	local_irq_restore(flags)	do { (void)(flags); } while (0)
+#define	nth_page(pg, i) \
+	((struct page *)((void *)(pg) + (i) * PAGESIZE))
+
```

Comment:
So I don't think this is the correct way to fix the PAGE_SHIFT. I was looking through and on AARCH64 this should be defined in asm/page.h so I'm trying to include it in abd.c as well. This is somehow getting pulled in on the x86 and x86_64 case but not the aarch64 case.